x-pack/platform/plugins/shared/cases/server/client/validators.ts

```diff
@@ -127,3 +128,14 @@ export const validateObservableTypeKeyExists = async (
     throw Boom.badRequest(`Invalid observable type, key does not exist: ${observableTypeKey}`);
   }
 };
+
+export const validateObservableValue = (
+  observableTypeKey: string | undefined,
+  observableValue: unknown
+) => {
+  const validationError = getValidatorForObservableType(observableTypeKey)(observableValue);
+
+  if (validationError) {
+    throw Boom.badRequest('Observable value is not valid for selected observable type.');
```

Comment:
super nit: May be adding key and value in the error message could be helpful?

Reply:
Just as an inline string?

Reply:
Yup, something like `'Observable value ${observableValue} is not valid for selected observable type ${observableTypeKey}.'`

Reply:
done, I will merge this pr after this the pipeline completes